### ISSUE TITLE
As a CAS user, I want to manage the list of normalized SWRS fuel names

### ISSUE THREAD

**Issue by Sriram-pc** [bcgov/cas-ggircs#384]
Description:

The list of fuels in the `swrs.fuel_carbon_tax_details` table is only editable by developers at the moment. This list contains the normalized fuel names, and connects them to the fuels defined in the carbon tax act. This feature would allow CAS staff to make updates without the development team's involvement.

This feature should provide a UI allowing CAS staff to manage the contents of the `swrs.fuel_carbon_tax_details` table 

Acceptance Criteria:
Given I am logged into the ggircs app
When I am on the landing page
Then there is a option to manage the normalized fuel types

Given I am in the fuel types management UI
When I first load the page
Then any fuels that are not associated to a normalized fuel type are clearly shown
and I can easily set the normalized fuel type for that fuel

Given I am in the fuel types management UI
When I want to find a fuel type
Then I can search the normalized fuel types

Given I am in the fuel types management UI
When I need to edit a normalized fuel type
Then I can make a change to the normalized fuel type

Given I am in the fuel types management UI
When I need to remove a fuel's association with a normalized fuel type
Then I can remove that fuel from the normalized fuel type
and it is then shown at the top of the page as an unassociated fuel

Development Checklist
* [x] UI has a design
* [x] user can navigate to a page to manage the fuel types
* [ ] fuel types without a normalized fuel type are shown at the top of the page
* [ ] fuel types without a normalized fuel type at the top of the page have a dropdown to select a normalized fuel type
* [ ] re-using the NAICS ui, clicking on a normalized fuel type shows the details and the associated fuels
* [ ] normalized fuel types can be searched
* [ ] normalized fuel types can be edited
* [ ] fuels can be removed from a normalized fuel type, doing so adds them to the list of fuels without a normalized fuel type at the top of the page
* [ ] Meets the DOD

**Comment by dleard** [comment]
Should this include a UI for assigning new swrs fuels (ie: new spellings / actual new fuels) to a normalized fuel type?

**Comment by matthieu-foucault** [comment]
I think so yes

**Comment by dleard** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Climate-Action-Secretariat-60ca4121764d710011481ca2/issues/bcgov/cas-ggircs/384?planning-poker) with ZenHub @AlexZorkin @matthieu-foucault @pbastia

**Comment by dleard** [comment]
Rough UI design![Screenshot from 2022-01-21 14-32-48.png](https://images.zenhubusercontent.com/5fc1589bc8692570749d2bed/9d18072f-eda6-4118-8ffa-244f7fc93bd2)

**Comment by dleard** [comment]
Something to consider with this work and the work in #385 (and any subsequent work that may touch the swrs schema) is that up until now, the swrs schema has been read-only, with no extra functions or mutations added. We really have two different types of data in this schema at the moment:
- table data that comes from ECCC and is transformed and loaded via materialized views
- table data that we have added from outside sources like the CTA, (ie: fuel_mapping and fuel_carbon_tax_details tables)

Our load process clones the current schema, loads updated data via materialized views into the clone, drops the current schema and renames the clone to match the original. We haven't had any custom mutations added to the swrs schema, but under the current load process those mutations would be lost in the drop.

I think it makes sense to add a new schema that won't be dropped for mutations and custom queries, but it also may make sense to move those contextual data tables like fuel_mapping into that new un-dropped schema as well. That would separate concerns, where the swrs schema would be purely data that comes from ECCC and context tables that we populate ourselves would live elsewhere. This would require a refactor in the code as well as manually fixing a lot of questions in metabase that would point to a non-existent table once those contextual tables were removed.
